### ISSUE TITLE
Example usage?

### ISSUE THREAD

**Issue by xiaodaigh** [quinnj/WeakRefStrings.jl#18]
Say I have an array of strings `x = ["abc", "def", "abc"]` how do I convert that to a vector of WeakRefStrings?

**Comment by quinnj** [comment]
Hi @xiaodaigh, thanks for posting the question. The accurate answer here is if you already have an array of strings, then WeakRefStrings don't really do you much good. WeakRefStrings are almost entirely an IO optimization when memory blocks representing strings are made available to a Julia process (e.g. reading a CSV file, transferring data from a database, etc.). In those cases, the IO process first starts w/ a chunk of memory and starts turning that memory into useful Julia structures; for strings, it can be a huge benefit to just "point" to that original memory block for a "string type", and allow the user to convert it to a proper Julia String later on as needed (in a lot of cases, the strings are never materialized).

So, in short, WeakRefStrings are a form of "lazy string construction" for Julia in certain IO contexts, but really aren't that useful otherwise. In fact, one of my goals for the package is to be almost entirely hidden for regular day-to-day users.

**Comment by quinnj** [comment]
(now, in theory, we could reimplement a full suite of string processing functions to operate directly on a WeakRefString's underlying memory, thus avoiding the String materialization at all, but that would be a non-trivial amount of work, so it's definitely future work at this point).
